[Bug 1150014] All unapproved revisions comment in Review Revision

safwanrahman opened the pull request:
I have tested in my local. and its working fine!
r?

mythmon commented:
This needs tests.

mythmon commented on kitsune/wiki/templates/wiki/review_revision.html:
```diff
@@ -16,6 +16,23 @@
 
+      {% if unreviewed_revision %}
+        <p>
+        <div class="unreviewed-revision"> {{ ngettext('Unreviewed Revision:', 'Unreviewed Revisions:', unreviewed_revision|length ) }} </div><br>
```
Since this is inside the `<p>`, it should be indented one step more. Additionally, there should not be a space around the `{{ }}`. Good job with `nggettext` by the way. The is exactly how it should be used.

mythmon commented on kitsune/wiki/templates/wiki/review_revision.html:
```diff
@@ -16,6 +16,23 @@
+              <br>
+              <a href="{{ url('wiki.review_revision', document.slug, revision.id) }}">{{ _('Review Revision {id}')|f(id=revision.id) }}</a>
+            </li>
+        {% endfor %}
```
The indentation here does not match the indentation of the opening `{% for %}` tag, on line 23.

mythmon commented on kitsune/wiki/views.py:
```diff
@@ -525,6 +525,20 @@ def review_revision(request, document_slug, revision_id):
         based_on_revs.values_list('creator__username', flat=True)))
 
+    # Get Unreviewed Revisions
+    unreviewed_revision = list(Revision.objects.filter(
```
Since this is multiple revisions, the name should be `unreviewed_revisions`, to denote that it is more than one item.

mythmon replied:
Also, generally calling `list()` on an ORM query is a sign that something isn't quite right. If you don't do the `if rev_in unreviewed_revision` thing below, you won't need to do `list()` here. That will be more efficient, as it will only get the first 5 from the DB, and not the entire list of every revision that matches these criteria.

mythmon commented on kitsune/wiki/views.py:
```diff
@@ -525,6 +525,20 @@ def review_revision(request, document_slug, revision_id):
+
+    # Remove the revision which is reviewing and get last 5 revision
+    if rev in unreviewed_revision:
+        unreviewed_revision.remove(rev)
```
This should be done with a `.exclude(id=rev.id)` clause in the ORM query above.

mythmon commented on kitsune/wiki/views.py:
```diff
@@ -525,6 +525,20 @@ def review_revision(request, document_slug, revision_id):
+    # Remove the revision which is reviewing and get last 5 revision
+    if rev in unreviewed_revision:
+        unreviewed_revision.remove(rev)
+    unreviewed_revision = unreviewed_revision[0:5]
```
When slicing, the first argument defaults to 0. In other words, this is better written as `unreviewed_revision = unreviewed_revision[:5]`.

mythmon commented:
I have a lot of little minor issues with this, but that is all they are: minor issues. This is actually quite a good PR, and I just want to make sure it fits our style guidelines and will be easy to work on in the future.

The tests you should write for this should generate a document and some revisions, and then check to see if the template contains the kinds of things you expect it to. If you need help with this, I can provide some examples of other code that does similar things. Ping me in irc, and I'll go find them.

safwanrahman commented:
@mythmon, I have added test and corrected other things as you mentioned! :smiley: :relaxed: 
Please regard only the last commit. other commits are just for updating the branch!
r?

mythmon commented:
Like before, it isn't really appropriate to merge master into PRs branches. I'll check the changes. Can you fix the merges by rebasing this branch?

safwanrahman commented:
@mythmon I have fixed the things as per you mentioned.
r?
After r+, I will remove the master branch things and squash it into one!

mythmon commented:
This looks fine. Please rebase this branch to not have any merge commits in it.

safwanrahman commented:
@mythmon, Take a little Glance and I think its ready to merge! :)
